src/compound_model/CompoundModelFactory.py

```diff
@@ -0,0 +1,403 @@
+        sample_metrics: list[Type[SampleMetric]],
+        compound_metrics: list[Type[CompoundMetric]],
+    ):
+        """
+        Constructs a CompoundModelFactory with the specified arguments.
+
+        Args:
+            controller (Controller): Controller of the CompoundModelFactory. Generator and Dataset will be inferred from
+                the controller.
+            filters (list[Type[Filter]]): Filters of the CompoundModelFactory, each filter in combination with the
+                controller is defined as a compound model.
+            sample_metrics (list[Type[SampleMetric]]): All sample metrics used to evaluate the model.
+            compound_metrics (list[Type[CompoundMetric]]): All compound metrics used to evaluate the model.
+        """
+        # Define key compound model components
+        self._controller = controller
+        self._generator = controller.get_generator()
+        self._filters = filters
+
+        # Define data objects
+        self._dataset = self._generator.get_dataset()
+        self._compound_model_name_prefix = "_".join(
+            [
+                controller.get_name(),
+                self._generator.get_name(),
+            ]
+        )
+        self._name = "_".join(
+            [COMPOUND_MODEL_FACTORY_PREFIX_NAME, self._compound_model_name_prefix]
+        )
+        self._population = Population(self._name)
+
+        # Define metric related objects, one manager for each compound model
+        self._smms = {
+            filter.get_name(): SampleMetricManager(
+                sample_metrics, self._population, self._dataset
+            )
+            for filter in self._filters
+        }
+        self._cmms = {
+            filter.get_name(): CompoundMetricManager(
+                compound_metrics,
+                self._population,
+                self._dataset,
+                self._smms[filter.get_name()],
+                controller,
+                filter.get_bit(FilterRegistry),
+            )
+            for filter in self._filters
+        }
+
+    def generate_custom_population(
+        self,
+        latent_codes: np.ndarray,
+        input: dict[str, np.ndarray],
+        pop_name: str,
+        append=True,
+    ):
+        """
+        Generates samples based on specified latent codes and input (which specify
+        attribute manipulations).
+
+        Samples will be saved under the specified population name.
+
+        An example of the input:
+
+        Let input be defined as:
+        `input_example = {"gender" : np.ndarray[-1, 1, -0.5], "age": np.ndarray[1,0,-1]}`
+        then input for image 1 should then be parsed as an image with gender = -1
+        and age = 1. Thus, `input_example` defines parameters for 3 images (length of the array).
+
+        Note that a value of 0 means that no changes for that parameter will be done for that image.
+
+        Note that the custom population is not meant to be evaluated using the `evaluate_compound_models()`.
```

Comment:
"...the custom population is never meant to be evaluated." maybe? idk if mentioning the function may be a bit misleading perhaps

Reply:
Made it more clear, i want to get the point across that you can actually calculate metrics for the custom population, just that it often doesn't make any sense :).